packages/cli/src/metadataGeneration/methodGenerator.ts

```diff
@@ -203,16 +205,16 @@ export class MethodGenerator {
       throw new GenerateMetadataError(`Only one SuccessResponse decorator allowed in '${this.getCurrentLocation()}' method.`);
     }
 
-    const [name, description, produces] = getDecoratorValues(decorators[0], this.current.typeChecker);
-    const examples = this.getMethodSuccessExamples();
-
-    const expression = decorators[0].parent as ts.CallExpression;
+    const [firstDecorator] = decorators;
+    const [name, description, produces] = getDecoratorValues(firstDecorator, this.current.typeChecker);
+    const expression = firstDecorator.parent as ts.CallExpression;
```

Comment:
Hi @WoH, I tried to include your suggestion on the code, please check again. I was not sure how to handle the scenario when _ts.isCallExpression_ is false.
Also, in the code _decorator.parent as ts.CallExpression;_ is used multiple times, let me know if you agree that I replace those by _ts.isCallExpression_ on a different PR.

Reply:
I know this used to be common, especially before type guards existed, but I encourage the safer style now.
When in doubt, my recommendation is to check and error if my expectations don't hold up (since we have a nice reporting that highlights the offending code and links to it). This gives us better visibility.

That said, holding you to that standard in light of the preceding code would not prevent errors, so let's tackle that issue another time.